src/modules/git_status.rs

```diff
@@ -181,21 +183,37 @@ impl<'a> GitStatusInfo<'a> {
 }
 
 /// Gets the number of files in various git states (staged, modified, deleted, etc...)
-fn get_repo_status(context: &Context) -> Option<RepoStatus> {
+fn get_repo_status(context: &Context, config: &GitStatusConfig) -> Option<RepoStatus> {
     log::debug!("New repo status created");
 
     let mut repo_status = RepoStatus::default();
-    let status_output = context.exec_cmd(
-        "git",
-        &[
-            OsStr::new("-C"),
-            context.current_dir.as_os_str(),
-            OsStr::new("--no-optional-locks"),
-            OsStr::new("status"),
-            OsStr::new("--porcelain=2"),
-            OsStr::new("--branch"),
-        ],
-    )?;
+    let mut args = vec![
+        OsStr::new("-C"),
+        context.current_dir.as_os_str(),
+        OsStr::new("--no-optional-locks"),
+        OsStr::new("status"),
+        OsStr::new("--porcelain=2"),
+    ];
+
+    // for performance reasons, only pass flags if necessary...
```

Comment:
Looks great to me! 👍 
Seems so obvious in hindsight! Thank you for tackling this 😄